tests/general/FileModeBearTest.py

```diff
@@ -0,0 +1,219 @@
+import os
+import platform
+import stat
+
+from queue import Queue
+
+from bears.general.FileModeBear import FileModeBear
+from coalib.testing.LocalBearTestHelper import LocalBearTestHelper
+from coalib.results.Result import RESULT_SEVERITY, Result
+from coalib.settings.Section import Section
+from coalib.settings.Setting import Setting
+
+
+FILE_PATH = os.path.join(os.path.dirname(__file__),
+                         'filemode_test_files', 'test_file.txt')
```

Comment:
the issue asked for "...on scripts" .  Why are you using `.txt` ?

Reply:
I guess having no extension at all would've been better. This is because the bear is supposed to check for all file permissions as mentioned [here](https://github.com/coala/coala-bears/issues/2370#issuecomment-401603938) and is not just restricted scripts.

Reply:
Start with the realistic scenario of scripts having `.sh` / `.bash` etc extensions and make sure you can get that working and tested problem, before trying extensionless voodoo.